src/main/java/tools/jackson/databind/introspect/AnnotatedMethod.java

```diff
@@ -21,19 +33,57 @@ public final class AnnotatedMethod
     /**********************************************************************
      */
 
-    public AnnotatedMethod(TypeResolutionContext ctxt, Method method,
-            AnnotationMap classAnn, AnnotationMap[] paramAnnotations)
+    private AnnotatedMethod(TypeResolutionContext ctxt, Method method,
+            AnnotationMap classAnn, AnnotationMap[] paramAnnotations, MethodHandles.Lookup lookup)
     {
         super(ctxt, classAnn, paramAnnotations);
         if (method == null) {
             throw new IllegalArgumentException("Cannot construct AnnotatedMethod with null Method");
         }
         _method = method;
+
+        MethodHandle invoker;
+        try {
+            invoker = lookup.unreflect(_method);
+        } catch (final IllegalAccessException e) {
+            invoker = MethodHandles.dropArguments(
+                    MethodHandles.throwException(Object.class, IllegalAccessException.class)
+                            .bindTo(e),
+                    0, Object.class);
+        }
+        _invoker = invoker;
+        boolean staticMethod = Modifier.isStatic(method.getModifiers());
+        int argCount = method.getParameterCount() + (staticMethod ? 0 : 1);
+        _nullaryInvoker = argCount > 0 ? null : invoker.asType(MethodType.methodType(Object.class));
+        _unaryConstructorInvoker = argCount != 1 ? null : invoker.asType(MethodType.methodType(Object.class, Object.class));
+    }
+
+    public static AnnotatedMethod of(TypeResolutionContext ctxt, Method method,
+            AnnotationMap classAnn, AnnotationMap[] paramAnnotations)
+    {
+        if (method == null) {
```

Comment:
Can just `Objects.requireNonNull()` since it's internal failure if called with null.